observatory-platform/observatory/platform/utils/test_utils.py

```diff
@@ -249,26 +257,61 @@ def add_connection(self, conn: Connection):
         self.session.add(conn)
         self.session.commit()
 
-    def run_task(self, dag: DAG, task_id: str, execution_date: pendulum.Pendulum) -> TaskInstance:
-        """ Run an Airflow task.
+    def run_task(self, task_id: str, dag: DAG = None, execution_date: pendulum.Pendulum = None) -> TaskInstance:
+        """Run an Airflow task.
 
         :param dag: the Airflow DAG instance.
         :param task_id: the Airflow task identifier.
         :param execution_date: the execution date of the DAG.
         :return: None.
         """
-
+        # If dag or execution date are not set, get corresponding values from dag run
+        if not (dag and execution_date):
+            if self.dag_run is None:
+                raise Exception(
```

Comment:
Not sure what the best way to exit / raise an exception is and make the test fail.

Reply:
It might be more appropriate to use a TypeError, e.g. https://stackoverflow.com/questions/59349532/what-is-the-standard-exception-for-a-missing-value-in-python